cmd/podman/run.go

```diff
@@ -61,7 +63,14 @@ func runCmd(c *cliconfig.RunValues) error {
 		return errors.Wrapf(err, "error creating libpod runtime")
 	}
 	defer runtime.DeferredShutdown(false)
-
 	exitCode, err = runtime.Run(getContext(), c, exitCode)
+	if c.Bool("rmi") {
+		imageName := c.InputArgs[0]
+		if newImage, err := runtime.NewImageFromLocal(imageName); err != nil {
```

Comment:
I don't think you should use `err` here?   If the runtime.Run command fails, you will destroy the error message on this line.